geniza/corpus/tests/test_sitemaps.py

```diff
@@ -0,0 +1,40 @@
+from geniza.corpus.models import Document
+from geniza.corpus.sitemaps import DocumentScholarshipSitemap, DocumentSitemap
+
+
+class TestDocumentSitemap:
+    def test_items(self, document):
+        # Ensure that documents are supressed if they aren't public
+        assert document.status == Document.PUBLIC
+        sitemap = DocumentSitemap()
+        assert len(sitemap.items()) == 1
+
+        document.status = Document.SUPPRESSED
+        document.save()
+        sitemap = DocumentSitemap()
+        assert len(sitemap.items()) == 0
+
+
+class TestDocumentScholarshipSitemap:
+    def test_items(self, document, footnote):
+        # Ensure that documents are supressed if they aren't public
+        document.status = Document.SUPPRESSED
+        document.save()
+        sitemap = DocumentScholarshipSitemap()
+        assert len(sitemap.items()) == 0
+
+        # Ensure that only documents with footnotes are returned
+        document.status = Document.PUBLIC
+        document.save()
+        sitemap = DocumentScholarshipSitemap()
+        assert len(sitemap.items()) == 1
+
+        document.footnotes.all().delete()
+        sitemap = DocumentScholarshipSitemap()
+        assert len(sitemap.items()) == 0
+
+    def test_location(self, document, footnote):
+        document.footnotes.add(footnote)
+        document.save()
+        sitemap = DocumentScholarshipSitemap()
+        assert sitemap.location(sitemap.items()[0]).endswith("scholarship/")
```

Comment:
please test with `reverse` and check the actual url

also test the last modified function, and priority if you add it based on zero footnotes

Reply:
33f34caf4ae834c00486002fce209754a99ccee5 ✅